### DISCUSSION TITLE
Is it possible to add the default plugins' config files to the /plugins directory?

### DISCUSSION THREAD

**Discussion by anasouardini** [LazyVim/LazyVim#1594] [Q&A]
As a new comer to Lazyvim, I find it very confusing when I try to alter the configuration of a default plugin.

I think it would be very simple and easy to figure out how to edit those plugins without having to google too much if the plugins and their default configs are available all in one directory `/plugins`.

I imagine that the config file of the plugin would also contain the option to enable it or not, instead of having to return and object from another file.

Just some thoughts, let me know what you think. Thanks for consideration.

**Comment by folke** [comment]
First of all, this is not an **issue**, so please use discussions for this.

Secondly, check the docs https://www.lazyvim.org/

It's all in there.

**Reply by anasouardini** [reply]
My bad, thanks for the heads-up.

I'm not sure what do you mean by linking the docs. As you already know, the documentation contains a lot of topics about Lazyvim, I don't know if I would go over all of it just to figure out what you can provide me in one paragraph!

Besides, I'm not sure I could find the answer to my question in there.

**Reply by dpetka2001** [reply]
Well the docs are there for a reason and you should at least read the bit about [plugins](https://www.lazyvim.org/configuration/plugins) which has the answer to what you're asking, how to add and customize plugins. I think if you take a look at the titles of the website, they are pretty self explanatory about what you should read.

**Reply by folke** [reply]
?
The docs are there for users to look at.

Why would I have to spend time to lookup the relevant sections and giving them to you?

Seriously. First a low effort issue and now telling me I should spend time doing stuff you can do yourself?

**Reply by anasouardini** [reply]
Oh, I see, my question is not clear.

What I meant is not to ask about adding/removing plugins, I'm suggesting that the default plugins that come with Lazyvim, could be added to the /plugins directory for noob-firendly modification. I've already read that page @dpetka2001, I appreciate your suggestion.

@folke Well, I thought you would already know without having to look it up. Sorry, that was not my intention.

"a low effort issue" ? Please tell me what I'm missing in the issue/discussion, I'll happily put the effort needed.

**Reply by folke** [reply]
The default options for all the plugins included in LazyVim are literally on the website.
"a low effort issue"  => you created an issue for something that should clearly be a discussion. You also didn't use one of the github issue templates I created specifically to prevent people posting issues like this

**Reply by anasouardini** [reply]
- ok, default options are in the docs, thanks. But why not have them in the `/plugins` dir, that's what I'm asking about.
- Yh, sorry for my short sightedness, i thought it's ok for it to be in the issues, I don't deal with opening issues that often. Also it's a suggestion, I didn't think it fits into the two provided templates.

Sorry if I'm wrong, but I'm sensing that my suggestion about changing the file system is not something you discuss in here.

Either way, thanks for your time, I'll try to gather all those defaults and add them to `/plugins` so that I don't need to hit google every now and then for one line of configuration, or slightly worse, read an entire page of docs.

**Reply by folke** [reply]
LazyVim is just a plugin like any other plugin installed with the lazy plugin manager.
Automatically adding those files will never happen, that makes little sense to me, since it would be impossible to properly make updates.

I would also highly advise against copying LazyVim's default options for every plugin in your own config.
This would mean you would be using a version of the defaults from the time you make the copy. So you would possibly lose out on fixes or better defaults later on.

To know what options a plugin has, you obviously **should** read its documentation. The LazyVim website only shows the defaults set by LazyVim, not all possible options. The same applies here for just blindly copying over all default options of a plugin from their readme/help.

**Reply by anasouardini** [reply]
Oh, I see, thanks for the clarification, I thought it's a simple problem while it's deeper than that.

I mostly get confused about, to which plugin does the feature belong in the first place, leaving me with really vague google queries that have to be modified as I learn more about the actual feature I'm trying to alter/remove.

For instance, I couldn't tell if the notification feature is a Lazyvim's feature or some notification plugin until I found this "fix":
```lua
return {
  "rcarriga/nvim-notify",
  opts = {
    level = 3,
    render = "minimal",
    stages = "static",
  },
}
```

I thought having those options included out of the box would make the process of modifying the config easier, but I think I'll just read more and adapt to it.

Thanks for taking the time to understand my newbie thoughts. You saved my evening from a lot of trouble.

**Comment by anasouardini** [comment]
Oh! `discussion/idea` is where this fits.
